src/theory/quantifiers/sygus/sygus_grammar_cons_new.h

```diff
@@ -0,0 +1,199 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Andrew Reynolds
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2023 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * Class for constructing inductive datatypes that correspond to
+ * grammars that encode syntactic restrictions for SyGuS.
+ */
+
+#include "cvc5_private.h"
+
+#ifndef CVC5__THEORY__QUANTIFIERS__SYGUS_GRAMMAR_CONS_NEW_H
+#define CVC5__THEORY__QUANTIFIERS__SYGUS_GRAMMAR_CONS_NEW_H
+
+#include <map>
+#include <vector>
+
+#include "expr/node.h"
+#include "expr/sygus_grammar.h"
+#include "options/options.h"
+
+namespace cvc5::internal {
+namespace theory {
+namespace quantifiers {
+
+/**
+ * Utility for constructing datatypes that correspond to syntactic restrictions.
+ */
+class SygusGrammarCons
+{
+ public:
+  /**
+   * Make the type corresponding to the default sygus grammar type returning
+   * range with bound variables bvl.
+   *
+   * @param opts The options, which impact grammar construction
+   * @param range The type of terms generated by the start symbol of the
+   * grammar.
+   * @param bvl The input variable list for the grammar (if it exists). If
+   * provided, each variable in this list is added as a non-terminal rule.
+   */
+  static TypeNode mkDefaultSygusType(const Options& opts,
+                                     const TypeNode& range,
+                                     const Node& bvl);
+  /**
+   * Make the type corresponding to the default sygus grammar type returning
+   * range with bound variables bvl, with a custom set of initial non-terminal
```

Comment:
Why non-terminal rules? Shouldn't they be terminal rules?

Reply:
Yes